numpy/f2py/f90mod_rules.py

```diff
@@ -161,8 +161,8 @@ def iadd(line, s=ihooks):
                 fargs.append('f2py_%s_getdims_%s' % (m['name'], n))
                 efargs.append(fargs[-1])
                 sargs.append(
-                    'void (*%s)(int*,int*,void(*)(char*,int*),int*)' % (n))
-                sargsp.append('void (*)(int*,int*,void(*)(char*,int*),int*)')
+                    'void (*%s)(int*,npy_intp*,void(*)(char*,npy_intp*),int*)' % (n))
+                sargsp.append('void (*)(int*,npy_intp*,void(*)(char*,npy_intp*),int*)')
                 iadd('\tf2py_%s_def[i_f2py++].func = %s;' % (m['name'], n))
                 fadd('subroutine %s(r,s,f2pysetdata,flag)' % (fargs[-1]))
```

Comment:
Thanks, yeah.  So the values passed are clearly `intp`, so the change is correct from the C-side and we could probably also just use `f2py_init_func` but maybe someone likes to be able to audit that it maches.

So the change is correct, and we can put it in.  I am still confused, the `f2pysetdata` takes care of `intp`'s size not being the default integer.  I would have guessed that `s` should be specified as that also, but I don't understand the following (or maybe it gets taken from `f2pysetdata` even?).

Reply:
I added the generated Fortran to the gist. I don’t know if that helps.

I’m not very familiar with Fortran, but I’ve been trying to dig into the generated code to understand the issue. `s` is defined in `f90mod_rules.py` as `np.intp().itemsize`. In the generated Fortran, it’s `integer(8) s(*)`. That should be the same size as `npy_intp`, right?